src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -496,10 +492,25 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
      */
     private def createExecutionContext(): ExecutionContextExecutor = {
       if (isParallel) {
-        val parallelThreads = settings.YparallelThreads.value
-        val threadPoolFactory = ThreadPoolFactory(Global.this, this)
-        val javaExecutor = threadPoolFactory.newUnboundedQueueFixedThreadPool(parallelThreads, "worker")
-        ExecutionContext.fromExecutorService(javaExecutor, _ => ())
+        if (settings.YparallelSequential) {
+          object SingleNewThreadExectuor extends Executor {
```

Comment:
I'm not sure cleanup (after `shutdown` will be working properly after this, but since it's only for local runs and it won't even be used for perf_tester we should be fine.)